google-cloud-bigquery/lib/google/cloud/bigquery/query_job.rb

```diff
@@ -106,6 +106,18 @@ def bytes_processed
         end
 
         ##
+        # Describes the execution plan for the query.
+        #
+        # @return [Array<Google::Cloud::Bigquery::QueryJob::Stage>]
+        #
+        def query_plan
+          return nil unless @gapi.statistics.query.query_plan
```

Comment:
That's right. I think we observed that a query job always has `statistics.query`, which is why it isn't guarded anywhere else.